### PR TITLE
Split off build parts for the CI

### PR DESCRIPTION
**Pull request by ctron**
This splits off part of the release build into re-usable components.

### CODE REVIEW

**Comment by ctron** [comment]
That should work as a pre-req for creating "latest" containers

File: .github/workflows/build-binary.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+      - name: Setup | Cache Cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: "${{ matrix.target }}-release${{ matrix.edition }}-${{ hashFiles('**/Cargo.lock') }}"
```
Comment (jcrossley3) [inline .github/workflows/build-binary.yaml:83]:
```suggestion
        uses: Swatinem/rust-cache@v2
```

Comment (ctron) [reply .github/workflows/build-binary.yaml:83]: Doesn't that conflict with the other cache? Let me read into the use of that.

Comment (ctron) [reply .github/workflows/build-binary.yaml]: Looks like that requires the use of the `shared-key` argument, which I added.

**Review by bobmcwhirter — APPROVED** [review]

**Pull request merged**